utils/driverless.c

```diff
@@ -125,6 +125,9 @@ copy_output(int read_ipp_fd,		/* I - IPP file descriptor */
           continue;
         snprintf(ippfind_output, MAX_OUTPUT_LEN, "%s", ptr);
         cupsArrayAdd(service_uri_list_ipps,ippfind_output);
+
+        free(ippfind_output);
+        ippfind_output = NULL;
```

Comment:
AFAIR cupsArrayAdd() only adds a pointer to the array and does not copy/strdup the string. So if you free the string immediately afterwards as you do here, you end up with an array of invalid pointers.
The freeing has be done when the after using the array, right before deleting it. Or you define the array with a freeing function, then deleting it will also free the content.

Reply:
@tillkamppeter Yes, you're right - my test sample was incorrect (I wasn't accessing the element from the array afterwards...), which led me to think the element needs to be freed before other iteration to prevent losing an allocated pointer.
@nidhijainpnp sorry for the noise about array elements - my proposed fix is incorrect, I'll define arrays with ```cupsArrayNew3()``` which allows to send freeing function as a parameter.

Though freeing ```service_uri``` string is only thing correct in the PR...

I'll be back in a while.